Update docstring of bkfact and related getindex.

tpapp opened the pull request:
To be compatible with 88e7fbcd002c21e6a03fe18643d83838cdbb31f7.

Also include the actual decomposition format. Fix missing permutation
matrix in the `getindex` docstring.

tpapp committed:
Update docstring of bkfact and related getindex.

To be compatible with 88e7fbcd002c21e6a03fe18643d83838cdbb31f7.

Also include the actual decomposition format. Fix missing permutation
matrix in the `getindex` docstring.

andreasnoack commented on base/linalg/bunchkaufman.jl:
```diff
@@ -41,16 +41,13 @@ function bkfact!(A::StridedMatrix{<:BlasFloat}, rook::Bool = false)
-`rook` is false, rook pivoting is not used.
-The following functions are available for
-`BunchKaufman` objects: [`size`](@ref), `\\`, [`inv`](@ref), [`issymmetric`](@ref), [`ishermitian`](@ref).
+Compute the Bunch-Kaufman [^Bunch1977] factorization of a symmetric or Hermitian matrix `A` as ``PUDU'P'`` or ``PLDL'P'``, depending on which triangle is stored in `A`, and return a `BunchKaufman` object.
```
The permutations are symmetric so you can leave out the apostrophe after the `P`s.

tpapp replied:
This bothered me too, for another reason: to be consistent with the use of permutation matrices, I think I should consider writing
```julia
LDL' = PAP'
```
but in this specific case `P'=P⁻¹=P`, so basically anything goes.

[R, for example](http://stat.ethz.ch/R-manual/R-patched/library/Matrix/html/BunchKaufman-methods.html), uses `A = P LDL' P'`, so that's one way to go.

I thought about mentioning that `P` is symmetric and its own inverse in the docstring.

What do you think? I like the symmetry of the formula, but I am happy to make any changes you suggest.

andreasnoack replied:
It is redundant and leaving out the apostrophe clearly shows the structure of the factorization. I'm fine with mentioning it explicitly in the text as well.

tpapp replied:
Done, thanks for the comments.

andreasnoack reviewed (commented):
Great. Thank for fixing this. Just a single minor change is needed before merging.

tpapp committed:
Merge branch 'master' of https://github.com/JuliaLang/julia into bkfact-docstrings

tpapp committed:
Removed transpose from second permutation matrix.

Change requested by @andreasnoack.

Also mention properties of permutation matrix in docstring.
